homeassistant/components/sensoterra/sensor.py

```diff
@@ -0,0 +1,146 @@
+    ),
+    "RSSI": SensorEntityDescription(
+        key="RSSI",
+        has_entity_name=True,
+        state_class=SensorStateClass.MEASUREMENT,
+        suggested_display_precision=0,
+        device_class=SensorDeviceClass.SIGNAL_STRENGTH,
+        native_unit_of_measurement=SIGNAL_STRENGTH_DECIBELS_MILLIWATT,
+        entity_category=EntityCategory.DIAGNOSTIC,
+    ),
+    "LASTSEEN": SensorEntityDescription(
+        key="LASTSEEN",
+        has_entity_name=True,
+        device_class=SensorDeviceClass.TIMESTAMP,
+        entity_category=EntityCategory.DIAGNOSTIC,
+        translation_key="last_seen",
+    ),
+}
+
+
+async def async_setup_entry(
+    hass: HomeAssistant, entry: ConfigEntry, async_add_devices: AddEntitiesCallback
+) -> None:
+    """Set up Sensoterra sensor."""
+
+    coordinator: SensoterraCoordinator = entry.runtime_data
+
+    def _add_devices(devices: dict[str, SensoterraSensor]) -> None:
+        async_add_devices(
+            [
+                SensoterraEntity(coordinator, sensor)
+                for sensor in devices.values()
+                if sensor.type in SENSORS
+            ]
+        )
+
+    coordinator.add_devices_callback = _add_devices
+
+    _add_devices(coordinator.data)
+
+
+class SensoterraEntity(CoordinatorEntity[SensoterraCoordinator], SensorEntity):
+    """Sensoterra sensor like a soil moisture or temperature sensor."""
+
+    def __init__(
+        self,
+        coordinator: SensoterraCoordinator,
+        sensor: SensoterraSensor,
+    ) -> None:
+        """Initialize entity."""
+        super().__init__(coordinator, context=sensor.id)
+
+        self.sensor_id = sensor.id
+        self._attr_unique_id = sensor.id
+
+        self.entity_description = SENSORS[sensor.type]
+
+        # Make sure {depth} placeholders gets substituted.
+        self._attr_translation_placeholders = {"depth": str(sensor.depth)}
+        if isinstance(self.entity_description.name, str):
+            self._attr_name = self.entity_description.name.format(depth=sensor.depth)
+
+        # Add soil type to certain sensors.
+        if sensor.soil is not None and sensor.type in ["MOISTURE", "SI"]:
+            self._attr_extra_state_attributes = {"soil_type": sensor.soil}
+
+        self.device_info = DeviceInfo(
+            identifiers={(DOMAIN, sensor.serial)},
+            name=sensor.name,
+            model=sensor.sku,
+            manufacturer="Sensoterra",
+            serial_number=sensor.serial,
+            suggested_area=sensor.location,
+            configuration_url=CONFIGURATION_URL,
+        )
+
+        if sensor.value is not None:
+            self._attr_native_value = sensor.value
+
+    @callback
+    def _handle_coordinator_update(self) -> None:
+        """Handle updated data from the coordinator."""
+        sensors: dict[str, SensoterraSensor] = self.coordinator.data
+        for sensor in sensors.values():
+            if sensor.id == self.sensor_id:
```

Comment:
the sensors is a dict, why do we need to loop and can't access the correct one directly